packages/opentelemetry-sdk-trace-base/src/SpanProcessor.ts

```diff
@@ -18,6 +18,11 @@ import { Context } from '@opentelemetry/api';
 import { ReadableSpan } from './export/ReadableSpan';
 import { Span } from './Span';
 
+/**
+ * Public interface of Span class
+ */
+export type WritableSpan = Pick<Span, keyof Span>;
```

Comment:
Are all fields of Span interface/type and not concrete class? I think tracestate might be the only problem?

Reply:
There is additionally `readonly resource: Resource;` which points to `class Resource`.

Reply:
We should probably also make those interfaces so we aren't bit by the same thing

Reply:
Do we have some naming guide regarding this? If we add interfaces for a lot classes where both of them are exported some consitency would be nice.
Or do you mean we should add an interface for `Resource` only in SDK not in `@opentelemetry/resources`?